File: LeetCode/Python Solutions/Count All Valid Pickup and Delivery Options/count.py
class Solution:
    def countOrders(self, n: int) -> int:
        # Define the modulo constant
        MOD = 10**9 + 7
        
        # Initialize the result to 1
        res = 1
        
        # Calculate the result using the formula
        for i in range(1, n + 1):
            res = (res * i * (2*i - 1)) % MOD
        
        return res

File: LeetCode/Java Solutions/Count All Valid Pickup and Delivery Options/count.java
class Solution {
    private static final int MOD = 1000000007;
    
    public int countOrders(int n) {
        long res = 1;
        
        // Calculate the result using the formula
        for (int i = 1; i <= n; i++) {
            res = (res * i * (2 * i - 1)) % MOD;
        }
        
        return (int) res;
    }
}
